common/Geometry2d/Rect.cpp

```diff
@@ -237,20 +237,17 @@ bool Rect::nearPoint(Point other, float threshold) const {
         return true;
     }
 
-    Point ur = Point(pt[1].x(), pt[0].y());
-    Point ll = Point(pt[0].x(), pt[1].y());
-    Segment edges[4] = {Segment(pt[0], ur), Segment(ur, pt[1]),
-                        Segment(pt[0], ll), Segment(ll, pt[1])};
+    double min_x = std::min(pt[0].x(), pt[1].x());
+    double max_x = std::min(pt[0].x(), pt[1].x());
+    double min_y = std::min(pt[0].y(), pt[1].y());
+    double max_y = std::min(pt[0].y(), pt[1].y());
 
-    // If any edge of this rect is near the point, then the point is near the
-    // rect.
-    for (Segment& edge : edges) {
-        if (edge.nearPoint(other, threshold)) {
-            return true;
-        }
-    }
+    double dx = std::max<double>({0.0, max_x - other.x(), other.x() - min_x});
+    double dy = std::max<double>({0.0, max_y - other.y(), other.y() - min_y});
 
-    return false;
+    double distance_to_point = std::sqrt(dx * dx + dy * dy);
+
+    return distance_to_point < threshold;
```

Comment:
I don't understand how this logic works, can you explain it?

Reply:
I'll add comments, but it's pretty simple. Because the box is axis-aligned, it's easy to find the minimum distance in the `x` and `y` directions separately (at most one of `x_max - x, 0, x - x_min` will be positive, where "positive" represents "outside").

Reply:
ok yeah after you fix the typo it makes sense.